fix(core/ui): minor visual tweaks

mmilata opened the pull request:
- centered title for coinjoin loader (closes #2946)
- make device label bold on label confirmation screen (closes #2953)

~~The second thing should be redone using FormattedText or similar after #2610 is merged, marking as draft until then.~~

<!--
If you are a core dev:

Don't forget to set up the fields:
- assign yourself
- set priority to same as original issue
- add PR to sprint
- if Draft -> "in progress"
- if final PR -> "needs review"

If you're an external contributor, you can ignore this text.
-->

Hannsek commented on core/src/apps/management/apply_settings.py:
```diff
@@ -153,8 +153,9 @@ async def _require_confirm_change_label(ctx: GenericContext, label: str) -> None
         "Device name",
-        description="Do you want to change device name to {}?",
-        description_param=label,
+        description="Do you want to change device name to",
```
Change it to "Change device name to…" please.

mmilata replied:
bf3adc743e81ec665638b9edd57a1981395279b6:

![Screenshot 2023-05-29 at 13-14-07 TT-device_tests-test_msg_applysettings py test_apply_settings](https://github.com/trezor/trezor-firmware/assets/85857/865506ce-8999-443e-b5cf-0fb4687326ff)

mmilata commented:
Rebased to master and changed to use FormattedText. Also did some refactoring in [d977e03](https://github.com/trezor/trezor-firmware/pull/3007/commits/d977e03bfe145bdd85d240167c600e5db7ec403d) so this will be more easily reviewed by individual commits.

master diff: https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/4488911152/artifacts/master_diff/index.html
differing screens: https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/4488911152/artifacts/master_diff/differing_screens.html

grdddj commented on core/embed/rust/src/ui/model_tt/layout.rs:
```diff
@@ -447,6 +457,32 @@ extern "C" fn new_confirm_action(n_args: usize, args: *const Obj, kwargs: *mut M
+        let end: StrBuffer = kwargs.get_or(Qstr::MP_QSTR_end, StrBuffer::empty())?;
+
+        let ops = OpTextLayout::new(theme::TEXT_NORMAL)
+            .text_normal(begin)
```
This configuration of normal-demibold-normal works for this one use-case, but cannot we make it more general - so that python can supply e.g. `list[tuple[str, int]]` with strings and font IDs?

We are already supplying and processing dynamic-sized lists in `confirm_properties`.

(One benefit would be also not having to define `MP_QSTR_{begin,middle,end}`)

Some quick pseudocode:
```rs
let items: Obj = kwargs.get(Qstr::MP_QSTR_items)?;
let mut iter_buf = IterBuf::new();
let iter = Iter::try_from_obj_with_buf(items, &mut iter_buf)?;
let mut ops = OpTextLayout::new(theme::TEXT_NORMAL);
for item in iter {
    let [string, font]: [Obj; 2] = iter_into_array(item)?;
    let string: StrBuffer = string.try_into()?;
    let font: i16 = font.try_into()?;
    match font {
        Font::NORMAL => {
            ops = ops.text_normal(string);
        },
        Font::BOLD => {
            ops = ops.text_bold(string);
        }
        ...
    }
}
```

mmilata replied:
Good idea, don't particularly enjoy `MP_QSTR_{begin,middle,end}` either. But I don't think we should expose the font choice to python - there's a precedent in `confirm_with_info`/`confirm_more` but I don't think we should add more as this goes against #2634.

In 03c7bfe293bab0ad5a499f7095846dae5cd87e6e I implemented this but with bool instead of font number, indicating whether that part of the text should be emphasized (by using demibold font on TT). This should work across hw models.

grdddj replied:
Good compromise, I agree that sending the `font ID` is unnecessary detail for `micropython`, sending just `is_emphasized` leaves it all to `Rust` to decide.

grdddj reviewed (commented):
Small comments on some namings and the limitations of `confirm_single` function, which could be made more general.

Functionality-wise it is OK.

Hannsek commented:
Can the buttons be changed here or should I open another issue? 
![image](https://github.com/trezor/trezor-firmware/assets/57008159/2e3e635a-7d06-4b4a-b520-026d89827bbc)

Hannsek commented:
Does this change the layout/copy/buttons for T2B1?

grdddj commented:
> Does this change the layout/copy/buttons for T2B1?

For `TR`, it changes only the wording of the `Change device name...` dialogue - see the bottom part of master diff - https://satoshilabs.gitlab.io/-/trezor/trezor-firmware/-/jobs/4488911152/artifacts/master_diff/differing_screens.html

mmilata commented on core/embed/rust/src/ui/component/text/formatted.rs:
```diff
@@ -14,19 +14,42 @@ use super::{
+            .layout_ops(self.char_offset, Offset::y(self.y_offset), sink)
+    }
+
+    fn align(&mut self, content_height: i16) {
```
9dcabd8216870f1891d511ae3109b11b5db6c7f6

Hannsek commented:
> For TR, it changes only the wording of the Change device name... dialogue 
Then I will create separate issue for TR.

grdddj reviewed (approved):
LGTM! Looking forward to using the vertically-alligned `FormattedText` in `TR`

bosomt commented:
**QA OK**

**Info:**
 - Suite version: desktop 23.6.1 (bddd516378beadfcff94523e7c94834ba14f147f)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuite/23.6.1 Chrome/112.0.5615.183 Electron/24.3.1 Safari/537.36
 - OS: MacIntel
 - Screen: 1440x900
 - Device: model T 2.6.1 Universal (revision c3f6e8f19fd647c73b414322e287ba23cf8a0c7c)
 - Transport: BridgeTransport 2.0.33
